### PR TITLE
Startup scripts: verify Java 8 (exactly), improve port/java verification messages.

### PR DESCRIPTION
**Pull request by gianm**
Java 11 compatibility isn't fully baked yet (users have reported various
issues on Java 11), so block startup with an error message unless Java 8
is found. Allow overriding this decision with an environment variable.

### CODE REVIEW

File: examples/bin/verify-default-ports [hunk 1]
```diff
@@ -21,12 +21,28 @@ use strict;
...
+  if (!bind($sock, sockaddr_in($port, INADDR_ANY))) {
+    print STDERR <<"EOT";
+Cannot start up because port $port is already in use.
+
+If you believe this check is in error, you can skip it using an
+environment variable:
+
+  export DRUID_SKIP_PORT_CHECK=1
+
+Otherwise, free up port $port and try again.
```
Comment (suneet-amp) [inline examples/bin/verify-default-ports]: nit: Would be nice to link to https://druid.apache.org/docs/latest/configuration/index.html for more details on changing the ports / what the different ports are used for.

File: examples/bin/verify-java [hunk 2]
```diff
@@ -20,14 +20,35 @@
 use strict;
 use warnings;
 
+sub fail_check {
+  my ($current_version) = @_;
+  my $current_version_text =
+    $current_version ? " Your current version is: $current_version." : "";
```
[the remark this answers is not in the document]
Comment (gianm) [reply examples/bin/verify-java]: Good idea, I added `Make sure that "java" is installed and on your PATH.`

File: examples/bin/verify-default-ports (hunk 1 above)
Comment (gianm) [reply examples/bin/verify-default-ports]: I made some adjustments, let me know what you think.

Comment (suneet-amp) [reply examples/bin/verify-default-ports]: Looks good!

**Comment by suneet-amp** [comment]
Thanks for this fix!

**Pull request merged**